app/assets/javascripts/core/comment_controller.coffee

```diff
@@ -2,10 +2,40 @@
 
 class Spaces.CommentsController extends Spaces.ViewController
 
-	initialize: ->
+	initialize: ([ @names ]) ->
 		($ 'body').on 'commentAdded', (event, data) => 
 			@addCommentToPost data
+		@addTagListener()
 
 	addCommentToPost: (data) =>
 		($ '.previous_comments').append data
-		($ '.comments form textarea').val ""
\ No newline at end of file
+		($ '.comments form textarea').val ""
+		($ '.comments form .textoverlay').html ""
+
+	addTagListener: =>
+		$.ajax
+			url: 'users/allnames'
+			type: 'GET'
+			success: (data) ->
```

Comment:
this is a non-trivial function (over 3 lines) so my recommendation would be to make it into a function or method, and then do something like: `success: (data) => @completeText(data)`